packages/types/api.d.ts

```diff
@@ -875,21 +880,21 @@ export declare class CstParser extends BaseParser {
   /**
    * Creates a Grammar Rule
    */
-  protected RULE(
+  protected RULE<ARGS extends unknown[]>(
```

Comment:
What is the purpose of "extends unknown" here?
- https://github.com/microsoft/TypeScript/issues/38456

Reply:
Would the generic argument gets resolved / inferred  correctly according to the provided implementation?

Meaning would a SUBRULE call that provides ARGS have type checks on the ARGS depending on which rule is invoked?

Reply:
extends "unknown array". This is needed to do `...args: ARGS`. ... which I wasn't doing. Should be fixed now.